### PR TITLE
feat: add quality of life features to the controllers

### PR DESCRIPTION
**Pull request by ahmedkhalf**
This PR aims to improve the quality of life for a user that is using the program. It will achieve this by:
- Correcting the spelling of the user if he is off by a few characters
- Enumerating the help of available commands using numbers
- Allow the user to call a command by its number

### Screenshot

image

### CODE REVIEW

**Review by ryanMMM — APPROVED** [review]
LGTM!

File: src/controllers/TerminalController.java [hunk 1]
```diff
@@ -75,6 +78,53 @@ private void exit() {
...
     }
 
+    private String processSpelling(String inputtedCommand) {
+        int maxDistance = (int) Math.ceil(inputtedCommand.length() / 8f);
+        LevenshteinDistance levenshteinDistance = new LevenshteinDistance();
+        int currentMin = Integer.MAX_VALUE;
+        String newCommand = inputtedCommand;
+        for (String command : AllCommands().keySet()) {
+            int dist = levenshteinDistance.apply(command, inputtedCommand);
+            if (dist <= maxDistance) {
```
Comment (DanielDervishi) [inline src/controllers/TerminalController.java]: This can be reduced to a single if statement

File: src/controllers/TerminalController.java [hunk 2]
```diff
@@ -75,6 +78,53 @@ private void exit() {
...
+            return inputtedCommand;
+        }
+
+        Integer numberCommand = processNumber(inputtedCommand);
+        if (numberCommand != null) {
+            return new ArrayList<>(AllCommands().keySet()).get(numberCommand);
+        }
+
+        String correctSpelling = processSpelling(inputtedCommand);
+        if (!Objects.equals(correctSpelling, inputtedCommand)) {
```
Comment (DanielDervishi) [inline src/controllers/TerminalController.java]: The if statement is un-necessary. Due to the first if statement in the method, if the inputted command has the correct spelling we just return the command associated with the input.

File: src/controllers/TerminalController.java (hunk 1 above)
Comment (ahmedkhalf) [reply src/controllers/TerminalController.java]: I am using the dist variable in the if statement body, therefore this cannot be reduced.

File: src/controllers/TerminalController.java (hunk 2 above)
Comment (ahmedkhalf) [reply src/controllers/TerminalController.java]: Done!

File: src/controllers/TerminalController.java (hunk 1 above)
Comment (ahmedkhalf) [reply src/controllers/TerminalController.java]: Oh never mind, I understand what you mean, highlighting line 89 too would have been great.

File: src/controllers/TerminalController.java [hunk 3]
```diff
@@ -95,9 +94,13 @@ private String processSpelling(String inputtedCommand) {
 
     private Integer processNumber(String inputtedCommand) {
         if (NumberUtils.isParsable(inputtedCommand)){
```
Comment (DanielDervishi) [inline src/controllers/TerminalController.java]: If statement un-necessary given the try-catch block. The return null in the catch block is redundant if the first if statement is removed.

**Review by DanielDervishi — APPROVED** [review]
Looks good to me.

**Pull request merged**